Add ensureScrollability option to scrollTo command

laistomazz opened the pull request:
<!-- 
Thanks for contributing!
Read our contribution guidelines here: 
https://github.com/cypress-io/cypress/blob/develop/CONTRIBUTING.md 
-->

<!-- Example: "Closes #1234" -->

- Closes https://github.com/cypress-io/cypress/issues/1924

### User facing changelog

You can now pass the `ensureScrollability: false` option to `.scrollTo()` to skip the scroll ability check of an element. Addresses https://github.com/cypress-io/cypress/issues/1924.

### Additional details

- I've used `ensureScrollability`, but it seems quite a long name for an option. Suggestions are welcome.
- I'll open a PR to update the documentation once or if the PR is approved and we agree on the option naming.

### How has the user experience changed?

Now users can use `scrollTo()` on an element that can have a variable height without getting an error.

#### Before

<img width="523" alt="Screen Shot 2020-07-20 at 10 56 11 AM" src="https://user-images.githubusercontent.com/1271364/87899740-14ad2800-ca78-11ea-9014-7165d3a23a98.png">

#### After

Passing `ensureScrollable: false`

<img width="522" alt="Screen Shot 2020-07-20 at 10 58 14 AM" src="https://user-images.githubusercontent.com/1271364/87899752-2393da80-ca78-11ea-9dde-cffb24fe5d90.png">

Passing incorrect arg to `ensureScrollable`

<img width="521" alt="Screen Shot 2020-07-20 at 10 58 34 AM" src="https://user-images.githubusercontent.com/1271364/87899760-2e4e6f80-ca78-11ea-8806-0338218a64a2.png">

Trying to scroll unscrollable el without `ensureScrollable: false`

<img width="493" alt="Screen Shot 2020-07-20 at 11 05 51 AM" src="https://user-images.githubusercontent.com/1271364/87900013-07dd0400-ca79-11ea-884c-e8f7571661e3.png">



### PR Tasks

<!-- 
These tasks must be completed before a PR is merged.
Delete tasks if they are not applicable. 
-->

- [x] Have tests been added/updated?
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? <!-- Link to PR here -->

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/cypress-io/cypress?pullRequest=8003) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/cypress-io/cypress?pullRequest=8003) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/cypress-io/cypress?pullRequest=8003) it.</sub>

laistomazz commented:
```html
<html>
<body>
</body>
</html>
```

```js
it('check for scrollability', () => {
    cy.visit('index.html')
    cy.scrollTo(0, 500)
})

it('skips scrollability check', () => {
    cy.visit('index.html')
    cy.scrollTo(0, 500, { ensureScrollability: false })
})

it('validates ensureScrollability value', () => {
    cy.visit('index.html')
    cy.scrollTo(0, 500, { ensureScrollability: 'force' })
})
```

![ezgif-3-7fba2f3d4c5b](https://user-images.githubusercontent.com/6031800/87662754-f63aea00-c762-11ea-81cd-137257e7eb05.gif)

jennifer-shehane commented on packages/driver/src/cy/commands/actions/scroll.js:
```diff
@@ -333,6 +338,10 @@ module.exports = (Commands, Cypress, cy, state) => {
       }
 
       const ensureScrollability = () => {
+        if (!options.ensureScrollability) {
```
Just some comments to clarify + link to original issue.

```suggestion
        // Some elements are not scrollable, user may opt out of error checking
        // https://github.com/cypress-io/cypress/issues/1924
        if (!options.ensureScrollability) {
```

laistomazz commented:
@jennifer-shehane Thanks for the review and the suggestions.

I initially used `force`, but then I found it a bit weird, because for `cy.click()`, for example, we use `{ force: true }` to click the element even if it's covered by something else (or for other reasons). In this case it will not scroll at all (?), the option would be for the case presented in the issue, for elements with variable content/height.

So I've changed it for `ensureScrollable` for now (until you discuss it with other team members).

I've also added the suggested change for the "not scrollable" error:
![Screenshot 2020-07-17 at 12 43 24](https://user-images.githubusercontent.com/6031800/87778950-4d58c180-c82c-11ea-9bb3-71707ccab9a2.png)

laistomazz commented on packages/driver/src/cy/commands/actions/scroll.js:
```diff
@@ -276,6 +277,10 @@ module.exports = (Commands, Cypress, cy, state) => {
       }
 
+      if (!_.isBoolean(options.ensureScrollability)) {
+        $errUtils.throwErrByPath('scrollTo.invalid_ensureScrollability', { args: { ensureScrollability: options.ensureScrollability } })
```
Done :) https://github.com/cypress-io/cypress/pull/8003/commits/7baf159c28c8c4ff114f0009bb1a7d0b873688b9
